api/provider/aws/service_list.go

```diff
@@ -9,102 +9,76 @@ import (
 )
 
 func (s *ServiceProvider) List() ([]models.ServiceSummary, error) {
-	clusterArns, err := s.listClusters()
+	clusterNames, err := listClusterNames(s.AWS.ECS, s.Config.Instance())
 	if err != nil {
 		return nil, err
 	}
 
-	clusterServices := map[*string][]*string{}
-
-	for _, clusterArn := range clusterArns {
-		clusterName := strings.Split(aws.StringValue(clusterArn), ":")[5]
-		clusterName = strings.Replace(clusterName, "cluster/", "", -1)
-
-		if !strings.HasPrefix(clusterName, s.Config.Instance()) {
-			continue
-		}
-
-		serviceArns, err := s.listServices(clusterArn)
-		if err != nil {
-			return nil, err
-		}
+	serviceARNs, err := s.listServiceARNsForClusterNames(clusterNames)
+	if err != nil {
+		return nil, err
+	}
 
-		clusterServices[clusterArn] = serviceArns
+	serviceIDs := make([]string, len(serviceARNs))
+	for i, serviceARN := range serviceARNs {
+		fqServiceID := strings.Split(serviceARN, "/")[1]
+		serviceID := delLayer0Prefix(s.Config.Instance(), fqServiceID)
+		serviceIDs[i] = serviceID
 	}
 
-	serviceSummaries := []models.ServiceSummary{}
-	for _, serviceArns := range clusterServices {
-		for _, serviceArn := range serviceArns {
-			ecsServiceID := s.serviceARNToECSServiceID(aws.StringValue(serviceArn))
-			service := &models.Service{
-				ServiceID: ecsServiceID,
-			}
-			s.populateSummariesTags(service, ecsServiceID)
+	return s.makeServiceSummaryModels(serviceIDs)
+}
 
-			summary := models.ServiceSummary{
-				ServiceID:       service.ServiceID,
-				ServiceName:     service.ServiceName,
-				EnvironmentID:   service.EnvironmentID,
-				EnvironmentName: service.EnvironmentName,
+func (s *ServiceProvider) listServiceARNsForClusterNames(clusterNames []string) ([]string, error) {
+	var serviceARNs []string
+	for _, clusterName := range clusterNames {
+		fn := func(output *ecs.ListServicesOutput, lastPage bool) bool {
```

Comment:
[super nit] we don't need to re-create `fn` for each environment. Also, it may help this function read a bit better to declare `fn` outside the for loop.

Reply:
Good point. :white_check_mark: